### PR TITLE
modeling complete

### PR DESCRIPTION
**Pull request by srPrajapat920**

### CODE REVIEW

File: app/models/loan.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Loan < ApplicationRecord
+  belongs_to :branch
+  belongs_to :user
+  validates :loan_type,:interest,:time_period,:amount,presence: true
```
Comment (ankitsahitya) [inline app/models/loan.rb:4]: add validations on loan type inclusion

File: app/models/transaction.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  validates :transaction_type,presence: true
+  validates :amount,numericality: {only_integer: true}
+  validates :transaction_type,inclusion: { in: %w(withdraw deposit)}
+  validate :amount_validate
+  validate :amount_limit
+  after_save :balance_update
+  private
+    def amount_validate
+      if self.amount <= 0
+      	errors.add(:amount,"invailid amount")
```
Comment (ankitsahitya) [inline app/models/transaction.rb]: correct indentation

File: app/models/transaction.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        errors.add(:amount,"insufficient balance")
+      end
+    end    	
+    def balance_update
+      if self.transaction_type == "withdraw"
+        left_balance = self.account.balance-self.amount
+        self.account.update_attributes(balance: left_balance)  
+      elsif self.transaction_type == 'deposit'
+        new_balance = self.account.balance+self.amount
+        self.account.update_attributes(balance: new_balance)
```
Comment (ankitsahitya) [inline app/models/transaction.rb]: optimise the code

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class User < ApplicationRecord
+  belongs_to :branch
+  has_many :accounts, dependent: :destroy
+  has_many :loans
+  validates :name,:father_name,:mother_name,:age,:address,:contact_no,presence: true
```
Comment (ankitsahitya) [inline app/models/user.rb:5]: add validation on age

Comment (srPrajapat920) [reply app/models/user.rb:5]: But we can create an account less than age 18..

Comment (ankitsahitya) [reply app/models/user.rb:5]: yes but can the age be less then 0
i can enter -1 age in it

Comment (srPrajapat920) [reply app/models/user.rb:5]: Got it

**Pull request closed without merging**